src/commands/compareReports.js

```diff
@@ -56,33 +56,40 @@ export default async function compareReports(
   if (dryRun) {
     log('Running in --dry-run mode -- no destructive commands will be issued');
   }
-  await Promise.all(
-    firstCompareResult.diffs.map(async ([before, after]) => {
-      const firstDiffDistance = await compareSnapshots({
-        before,
-        after,
-        endpoint,
-        compareThreshold,
-      });
-      if (!firstDiffDistance) {
-        log(
-          `✓ ${after.component} - ${after.variant} - ${
-            after.target
-          } - diff is within threshold`,
-        );
-        if (!dryRun) {
-          await ignore({ before, after, apiKey, apiSecret, endpoint });
+
+  const diffsClone = firstCompareResult.diffs.slice(0);
+  let batch;
+  // eslint-disable-next-line no-cond-assign
+  while ((batch = diffsClone.splice(0, 10)).length > 0) {
```

Comment:
Why batch here?

Reply:
To avoid kicking off thousands of image fetches at the same time. By doing 10 at a time we don't risk overloading I/O and memory.